src/Workspaces/Core/Portable/Formatting/Context/FormattingContext.cs

```diff
@@ -31,6 +31,7 @@ internal partial class FormattingContext
         private readonly ContextIntervalTree<IndentationData> _indentationTree;
         private readonly ContextIntervalTree<SuppressWrappingData> _suppressWrappingTree;
         private readonly ContextIntervalTree<SuppressSpacingData> _suppressSpacingTree;
+        private readonly ContextIntervalTree<SuppressSpacingData> _suppressFormattingTree;
```

Comment:
so, adding suppression rule that suppress "space and wrapping" didn't work? is that why suppress formatting tree is needed?

is adding a rule provider that goes through trivia of given span and find disable pragma and add SuppressOperation with options SupressOption.NoSpacing | SupressOption.NoWrapping didn't work? if that didn't work, can we improve that part rather than adding supressFormattingTree?

Reply:
in my mind, this pragma is basically setting a span in the tree that we don't allow to change any space or line changes. nothing special compared to array initializer and etc.

ex) http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp.Workspaces/Formatting/Rules/SuppressFormattingRule.cs,319

the above still format when construct is on single line, but formatter support (in theory :)) completely suppressing every change.

Reply:
I wasn't able to find a way to use those without disrupting a bunch of other features. It seemed to me like it should work, but I wasn't having success.

Reply:
was the issue of how rule is added? or how it is applied?